d3dcore/include/d3dcore/d3d_utils.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <exception>
+#include <stdexcept>
+#include <string>
+#include <vector>
+#include <Windows.h>
+
+namespace d3dcore {
+
+	
+		std::string WChar_to_string(WCHAR* wChar) {
```

Comment:
In general, try not to mix WinAPI and standard C++ methods. There are ways to convert from wchar_t to char in standard library